File: JavaScript/7js 2/nbDig.js
/*
|--------------------------------------------------------------------------
| Count the Digit
|--------------------------------------------------------------------------
|
*/

function nbDig(n, d) {
   let res = 0
   for (let i = 0; i <= n; i++) {
      res += String(i ** 2).split('').filter(i => i == String(d)).length;
   }
   return res
}


const nbDig = (n, d) => Array.from(Array(n + 1), (e, i) => i * i)
   .join('')
   .match(new RegExp(d, 'g'))
   .length;


function nbDig(n, d) {
   let o = '';
   for (let i = 0; i <= n; i++) {
      o += Math.pow(i, 2);
   }
   return o.split(d).length - 1
}


const nbDig = (n, d) => {
   let re = new RegExp(d, 'g');
   return Array(...Array(n + 1))
      .reduce((s, v, i) => s + ((i * i + '').match(re) || []).length, 0);
}


console.log(nbDig(5750, 0), 4700)
console.log(nbDig(11011, 2), 9481)
console.log(nbDig(12224, 8), 7733)
console.log(nbDig(11549, 1), 11905)

File: Python/7py 2/nb_dig.py
"""
|--------------------------------------------------------------------------
| Count the Digit
|--------------------------------------------------------------------------
|
"""

def nb_dig(n, d):
   res = 0
   for i in range(n+1):
      res += len(list(filter(lambda i: i == str(d) , str(i**2))))
   return res


def nb_dig(n, d):
   return sum(str(i*i).count(str(d)) for i in range(n+1))


def nb_dig(n, d):
   return ''.join(str(n * n) for n in range(n + 1)).count(str(d))


def nb_dig(n, d):
   return sum([str(x**2).count(str(d)) for x in range(n+1)])


print(nb_dig(5750, 0), 4700)
print(nb_dig(11011, 2), 9481)
print(nb_dig(12224, 8), 7733)
print(nb_dig(11549, 1), 11905)
print(nb_dig(25, 1), 11)